sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/MockVCIssuancePlugin.java

```diff
@@ -0,0 +1,255 @@
+        vEngine.init();
+
+        //Validate all the supported VC
+        String supportedCredentialTypes =env.getProperty("mosip.esignet.vciplugin.sunbird-rc.supported-credential-types");
+        if (supportedCredentialTypes == null || supportedCredentialTypes.isEmpty()) {
+            throw new IllegalArgumentException("Property mosip.esignet.vciplugin.sunbird-rc.supported-credential-types is not set.");
+        }
+        String[] credentialTypes = supportedCredentialTypes.split(",");
+        for (String credentialType : credentialTypes) {
+            validatePropertyForCredentialType(credentialType.trim());
+        }
+    }
+
+    @Override
+    public VCResult<JsonLDObject> getVerifiableCredentialWithLinkedDataProof(VCRequestDto vcRequestDto, String holderId, Map<String, Object> identityDetails) throws VCIExchangeException {
+        validateProperties();
+        return null;
+    }
+
+    @Override
+    public VCResult<String> getVerifiableCredential(VCRequestDto vcRequestDto, String holderId, Map<String, Object> identityDetails) throws VCIExchangeException {
+        return null;
+    }
+
+
+
+    private void validatePropertyForCredentialType(String credentialType) {
+        validateProperty("mosip.esignet.vciplugin.sunbird-rc.credential-type." + credentialType + ".template-url");
+        validateProperty("mosip.esignet.vciplugin.sunbird-rc.credential-type." + credentialType + ".registry-get-url");
+        validateProperty("mosip.esignet.vciplugin.sunbird-rc.credential-type." + credentialType + ".cred-schema-id");
+        validateProperty("mosip.esignet.vciplugin.sunbird-rc.credential-type." + credentialType + ".cred-schema-version");
+        validateProperty("mosip.esignet.vciplugin.sunbird-rc.credential-type." + credentialType + ".static-value-map.issuerId");
+        //TODO download  the templete
+        //TODO validate  the templete as JSON-ld
+        //Todo use Map of Map to cache configuration
+
+        String templateUrl = env.getProperty("mosip.esignet.vciplugin.sunbird-rc.credential-type." + credentialType + ".template-url");
+        downloadAndValideateTemplate(templateUrl,credentialType);
+        //validateJsonLd(template);
+        //credentialTypeTemplates.put(credentialType, template);
+
+    }
+
+    private void validateProperty(String propertyName) {
+        String propertyValue = env.getProperty(propertyName);
+        if (propertyValue == null || propertyValue.isEmpty()) {
+           throw new IllegalArgumentException("Property " + propertyName + " is not set.");
+        }
+    }
+
+    private void downloadAndValideateTemplate(String templateUrl,String credentialType){
+
+        //String templete=wrt.toString();
+        try{
+            Template t = vEngine.getTemplate(templateUrl);
+            VelocityContext context = new VelocityContext();
+            StringWriter wrt = new StringWriter();
+            t.merge(context,wrt);
+
+            System.out.println(wrt.toString());
```

Comment:
We should not use SOP statements in the code, we should use only loggers

Reply:
don't need to review the mockVCI yet it is steel not finished

Reply:
Ideally it should not be in the PR, since it is reverted, resolving this.